compiler/rustc_resolve/src/lib.rs

```diff
@@ -1722,13 +1722,6 @@ impl<'a> Resolver<'a> {
         debug!("resolve_crate_root({:?})", ident);
         let mut ctxt = ident.span.ctxt();
         let mark = if ident.name == kw::DollarCrate {
-            // When resolving `$crate` from a `macro_rules!` invoked in a `macro`,
-            // we don't want to pretend that the `macro_rules!` definition is in the `macro`
-            // as described in `SyntaxContext::apply_mark`, so we ignore prepended opaque marks.
-            // FIXME: This is only a guess and it doesn't work correctly for `macro_rules!`
-            // definitions actually produced by `macro` and `macro` definitions produced by
-            // `macro_rules!`, but at least such configurations are not stable yet.
-            ctxt = ctxt.normalize_to_macro_rules();
```

Comment:
Note: I removed this because it didn't change the behavior of any test (in the default compiler mode `x.py test`), not because removing it was necessary.

In general, I'm not certain how much of this arm is still necessary now that we're directly giving `$crate` a new mark when gluing it but at the very least removing the branch entirely didn't seem to work.